commcare/src/main/java/org/motechproject/commcare/pull/CommcareFormImporterImpl.java

```diff
@@ -64,6 +65,20 @@ public int countForImport(Range<DateTime> dateRange, String configName) {
     }
 
     @Override
+    public boolean checkFormIdForImport(String formId, String configName) {
+        LOGGER.info("Checking form with id {} [config: {}]", formId, configName);
+        try {
+            CommcareForm form = formService.retrieveForm(formId, configName);
+            LOGGER.info("Form with id {} exists", form.getId());
+            return true;
+        } catch (NullPointerException e) {
```

Comment:
catching NPE is always bad idea, you should rather modificate retrieveForm method from formService to return null if form does not exist and then "return form != null'